packages/engine/Source/Core/Resource.js

```diff
@@ -753,8 +725,16 @@ Resource.prototype.retryOnError = function (error) {
  */
 Resource.prototype.clone = function (result) {
   if (!defined(result)) {
-    result = new Resource({
+    return new Resource({
       url: this._url,
+      queryParameters: this.queryParameters,
+      templateValues: this.templateValues,
+      headers: this.headers,
+      proxy: this.proxy,
+      retryCallback: this.retryCallback,
+      retryAttempts: this.retryAttempts,
+      request: this.request.clone(),
+      parseUrl: false,
```

Comment:
`getDerivedResource` is also a heavily used function. Is it possible to use the same optimization in some cases there?